core/src/com/bot4s/telegram/clients/SttpClient.scala

```diff
@@ -76,8 +77,6 @@ class SttpClient[F[_]](token: String, telegramHost: String = "api.telegram.org")
       .response(asJson[Response[R]])
       .send[F]()
 
-    response
-      .map(_.unsafeBody)
-      .map(processApiResponse[R])
+    monadError.flatMap(monadError.map(response)(_.unsafeBody))(t => monadError.fromTry(Try(processApiResponse[R](t))))
```

Comment:
That looks good, however, could you please create an intermediate variable for readability.

```scala
   val extractedBody = monadError.map(response)(_.unsafeBody)
    monadError
      .flatMap(extractedBody)(t => monadError.fromTry(Try(processApiResponse[R](t))))
```

Thanks !
Also, could you provide a setup in which such an error could happen ? (we don't have tests for this class, but I would like to keep track of this kind of issue for the future)

Reply:
I'll try to get one during the week, have a busy day tomorrow and its already late now. The case where this happens is basically anywhere where the telegram api returns an error, e.g. as in my repo, when the bot tries to create an invite link, but has not the right permissions in the room for that, then telegram-api returns a 400.